src/components/TextField/TextField.tsx

```diff
@@ -103,6 +106,8 @@ export interface BaseProps {
   showCharacterCount?: boolean;
   /** Determines the alignment of the text in the input */
   align?: Alignment;
+  /** Callback when clear button is clicked */
+  onClearButtonClick?(id: string): void;
```

Comment:
@andrewmcgov I made a change here that I think makes a lot of sense. Instead of using the `onChange` handler passed as a prop to handle the clear button click, we're leaving that implementation up to the consumer. This is more in line with how we think about handlers when a component needs to have its state managed externally.

This also makes these changes a lot easier to cover with testing.

What do you think?

Reply:
@danrosenthal I like this - definitely solves the testing problem I was running into. It also lets a consumer do something like `setState({value: 0})` for number inputs, which could be really handy.

I wonder if we should add a default `clearInput` function to the textfield that clears the value to `''` if `onClearButtonClick` is `null`? 

This way adding a clearButton is as easy as passing a single prop, but it still gives the consumer the option to set up their own behaviour if they want to. 

I'm also fine leaving this the way it is, since you mentioned this is more in line with how we would normally set up a component with external state management.

Reply:
> I wonder if we should add a default clearInput function to the textfield that clears the value to '' if onClearButtonClick is null?
> 
> This way adding a clearButton is as easy as passing a single prop, but it still gives the consumer the option to set up their own behaviour if they want to.
> 
> I'm also fine leaving this the way it is, since you mentioned this is more in line with how we would normally set up a component with external state management.

This would be tricky given that the value state is managed externally from the component. The click on the clear button being a noop I think is fine and would signal to the consumer that they need to add it.